pallets/common/src/traits/identity.rs

```diff
@@ -251,6 +249,9 @@ decl_event!(
 
         /// All Secondary keys of the identity ID are unfrozen.
         SecondaryKeysUnfrozen(IdentityId),
+
+        /// An unexpected error happened that should be investigated.
+        UnexpectedError(Option<DispatchError>),
```

Comment:
iirc we wanted a special pallet for this.
Throwing it into the identity pallet-ish seems somewhat random, and also has the disadvantage of the unncessary `T` type parameter caused by the two projections in the `where`.
Also, what's with the `Option<_>`?

Reply:
The identity pallet is used in all our pallets anyway. That's why I reused it. I can add a `polymesh_global` and link it to all our pallets but it doesn't seem to be of much use unless we add more global functions.

The `Option` case is for when the error is completely unexpected and there's no relevant DispatchError. That being said, I doubt we'll ever get a situation like that in our codebase.

Reply:
I would prefer introducing a different event type, in this crate, just in a different module, and then using that, primarily so that we can avoid the type parameter and improve usage ergonomics. You might even get away without `decl_module!` etc. by just having `GlobalErr: global_err::{Event},` in `construct_runtime! { ... }`.